npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb

```diff
@@ -468,13 +481,33 @@ def lockfiles_for_path(lockfiles:, path:)
         def run_checker(path:, version:)
           # If there are both yarn lockfiles and npm lockfiles only run the
           # yarn updater
-          if lockfiles_for_path(lockfiles: dependency_files_builder.yarn_locks, path: path).any?
-            return run_yarn_checker(path: path, version: version)
+          lockfiles = lockfiles_for_path(lockfiles: dependency_files_builder.yarn_locks, path: path)
+          if lockfiles.any?
+            if Helpers.yarn_berry?(lockfiles.first)
+              return run_yarn_berry_checker(path: path, version: version)
+            else
+              return run_yarn_checker(path: path, version: version)
+            end
           end
 
           run_npm_checker(path: path, version: version)
         end
 
+        def run_yarn_berry_checker(path:, version:)
+          SharedHelpers.with_git_configured(credentials: credentials) do
+            Dir.chdir(path) do
+              output = Helpers.run_yarn_command("yarn add #{dependency.name}@#{version}#{Helpers.yarn_berry_args}")
+              if output.include?("YN0060")
+                raise SharedHelpers::HelperSubprocessFailed.new(
+                  message: output,
+                  error_class: "JSON::ParserError",
```

Comment:
Why are we raising a `JSON::ParseError` in this scenario?

Reply:
Lack of knowledge about other error times that might be more appropriate :)

Reply:
https://github.com/dependabot/dependabot-core/blob/main/common/lib/dependabot/errors.rb is where all the top-level errors that are handled in the backend service are defined. But looking at this closer, it seems like the `error_class` is not really used here, since we catch all `HelperSubprocessFailed` errors in the calling code. Seems like we can just leave it out here?